### PR TITLE
Codespaces setup

### PR DESCRIPTION
**Pull request by GrantBirki**
# CodeSpaces Setup

This pull request sets up necessary files and configuration to use GitHub CodeSpaces with Ersilia :rocket:

---

Resolves: https://github.com/ersilia-os/ersilia/issues/698

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+RUN apt-get install -y tzdata
+RUN apt-get install -y build-essential wget git curl zip docker
+
+RUN wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.11.0-Linux-x86_64.sh
+RUN chmod +x Miniconda3-py37_4.11.0-Linux-x86_64.sh
+RUN bash Miniconda3-py37_4.11.0-Linux-x86_64.sh -b
+
+# Setup git-lfs
+RUN eval "$(/root/miniconda3/bin/conda shell.bash hook)" &&\
```
Comment (GrantBirki) [inline .devcontainer/Dockerfile:13]:
```suggestion
RUN eval "$(/root/miniconda3/bin/conda shell.bash hook)" && \
```

**Comment by Lehcar** [comment]
Ok, so far, I'm able to fetch the demo malaria model (eos7pnm), but when I serve the model I see the following error:
```
Traceback (most recent call last):
  File "/root/miniconda3/envs/ersilia/bin/ersilia", line 33, in <module>
    sys.exit(load_entry_point('ersilia', 'console_scripts', 'ersilia')())
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 138, in wrapper
    return func(*args, **kwargs)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 115, in wrapper
    return_value = func(*args, **kwargs)
  File "/root/miniconda3/envs/ersilia/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 99, in wrapper
    return func(*args, **kwargs)
  File "/ersilia/ersilia/cli/commands/serve.py", line 28, in serve
    mdl = ErsiliaModel(
  File "/ersilia/ersilia/core/model.py", line 114, in __init__
    self.autoservice = AutoService(
  File "/ersilia/ersilia/serve/autoservice.py", line 34, in __init__
    service_class_file = os.path.join(
  File "/root/miniconda3/envs/ersilia/lib/python3.10/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

I don't think this is related to the codespace setup though

**Comment by Lehcar** [comment]
Seeing this error now:
```bash 
21:59:50 | DEBUG    | Service class: conda
21:59:50 | DEBUG    | Model successfully initialized in sniffer
21:59:50 | DEBUG    | No built-in example available. Generating a test one.
21:59:50 | DEBUG    | Reading card from eos1086
21:59:50 | DEBUG    | Reading shape from eos1086
21:59:50 | DEBUG    | Input Shape: List
21:59:50 | DEBUG    | Input type is: text
21:59:50 | DEBUG    | Input shape is: List
21:59:50 | DEBUG    | Importing module: .types.text
Traceback (most recent call last):
  File "/root/miniconda3/bin/ersilia", line 33, in <module>
    sys.exit(load_entry_point('ersilia', 'console_scripts', 'ersilia')())
  File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 138, in wrapper
    return func(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 115, in wrapper
    return_value = func(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 99, in wrapper
    return func(*args, **kwargs)
  File "/ersilia/ersilia/cli/commands/fetch.py", line 73, in fetch
    _fetch(mf, model_id)
  File "/ersilia/ersilia/cli/commands/fetch.py", line 12, in _fetch
    mf.fetch(model_id)
  File "/ersilia/ersilia/hub/fetch/fetch.py", line 247, in fetch
    self._fetch_not_from_dockerhub(model_id=model_id)
  File "/ersilia/ersilia/hub/fetch/fetch.py", line 201, in _fetch_not_from_dockerhub
    self._sniff()
  File "/ersilia/ersilia/hub/fetch/fetch.py", line 127, in _sniff
    sn = ModelSniffer(self.model_id, self.config_json)
  File "/ersilia/ersilia/hub/fetch/actions/sniff.py", line 69, in __init__
    eg = ExampleGenerator(model_id, config_json=config_json)
  File "/ersilia/ersilia/io/input.py", line 180, in __init__
    self.IO = BaseIOGetter(config_json=config_json).get(model_id)
  File "/ersilia/ersilia/io/input.py", line 68, in get
    return self._get_from_model(model_id=model_id)
  File "/ersilia/ersilia/io/input.py", line 54, in _get_from_model
    return importlib.import_module(module, package="ersilia.io").IO(
  File "/root/miniconda3/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'ersilia.io.types.text'
```

**Comment by carcablop** [comment]
> Seeing this error now when trying to run `ersilia fetch eos1086`:
> 
> ```shell
> 21:59:50 | DEBUG    | Service class: conda
> 21:59:50 | DEBUG    | Model successfully initialized in sniffer
> 21:59:50 | DEBUG    | No built-in example available. Generating a test one.
> 21:59:50 | DEBUG    | Reading card from eos1086
> 21:59:50 | DEBUG    | Reading shape from eos1086
> 21:59:50 | DEBUG    | Input Shape: List
> 21:59:50 | DEBUG    | Input type is: text
> 21:59:50 | DEBUG    | Input shape is: List
> 21:59:50 | DEBUG    | Importing module: .types.text
> Traceback (most recent call last):
>   File "/root/miniconda3/bin/ersilia", line 33, in <module>
>     sys.exit(load_entry_point('ersilia', 'console_scripts', 'ersilia')())
>   File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
>     return self.main(*args, **kwargs)
>   File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1055, in main
>     rv = self.invoke(ctx)
>   File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/root/miniconda3/lib/python3.10/site-packages/click/core.py", line 760, in invoke
>     return __callback(*args, **kwargs)
>   File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 138, in wrapper
>     return func(*args, **kwargs)
>   File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 115, in wrapper
>     return_value = func(*args, **kwargs)
>   File "/root/miniconda3/lib/python3.10/site-packages/bentoml/cli/click_utils.py", line 99, in wrapper
>     return func(*args, **kwargs)
>   File "/ersilia/ersilia/cli/commands/fetch.py", line 73, in fetch
>     _fetch(mf, model_id)
>   File "/ersilia/ersilia/cli/commands/fetch.py", line 12, in _fetch
>     mf.fetch(model_id)
>   File "/ersilia/ersilia/hub/fetch/fetch.py", line 247, in fetch
>     self._fetch_not_from_dockerhub(model_id=model_id)
>   File "/ersilia/ersilia/hub/fetch/fetch.py", line 201, in _fetch_not_from_dockerhub
>     self._sniff()
>   File "/ersilia/ersilia/hub/fetch/fetch.py", line 127, in _sniff
>     sn = ModelSniffer(self.model_id, self.config_json)
>   File "/ersilia/ersilia/hub/fetch/actions/sniff.py", line 69, in __init__
>     eg = ExampleGenerator(model_id, config_json=config_json)
>   File "/ersilia/ersilia/io/input.py", line 180, in __init__
>     self.IO = BaseIOGetter(config_json=config_json).get(model_id)
>   File "/ersilia/ersilia/io/input.py", line 68, in get
>     return self._get_from_model(model_id=model_id)
>   File "/ersilia/ersilia/io/input.py", line 54, in _get_from_model
>     return importlib.import_module(module, package="ersilia.io").IO(
>   File "/root/miniconda3/lib/python3.10/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
> ModuleNotFoundError: No module named 'ersilia.io.types.text'
> ```

This issue is because Ersilia doesn't accept text as input yet. This is being worked on here. https://github.com/ersilia-os/ersilia/issues/603

**Review by miquelduranfrigola — APPROVED** [review]
This looks good

**Comment by Lehcar** [comment]
Just to triple check, I ran the commands below in a codespace to test the molecular-weight model and it worked :] 
This is the model [that we test in the "Deploy and test ersilia on PR" action](https://github.com/ersilia-os/ersilia/blob/master/.github/workflows/pr_check.yml#L42-L48).

```bash
  source activate
  ersilia -v fetch molecular-weight --from_github
  echo "Serving molecular-weight model."
  ersilia serve molecular-weight
  ersilia run -i "CC(=O)OC1=CC=CC=C1C(=O)O" | grep "180.15899"
```

**Pull request merged**